packages/components/src/visually-hidden/index.js

```diff
@@ -8,13 +8,25 @@ import classnames from 'classnames';
  */
 import { renderAsRenderProps } from './utils';
 
+/**
+ * @template {keyof JSX.IntrinsicElements | import('react').JSXElementConstructor<any>} T
+ * @typedef OwnProps
+ * @property {T} [as='div'] Component to render, e.g. `"div"` or `MyComponent`.
+ */
+
+/**
+ * @template {keyof JSX.IntrinsicElements | import('react').JSXElementConstructor<any>} T
+ * @typedef {OwnProps<T> & import('react').ComponentProps<T>} Props
+ */
+
```

Comment:
Nice! I was taking a look at this last night and having a hard time wrapping my head around how to type this. Good work!